app/services/ClaimToAdjustService.scala

```diff
@@ -33,6 +34,42 @@ class ClaimToAdjustService @Inject()(val financialDetailsConnector: FinancialDet
                                      implicit val dateService: DateServiceInterface)
                                     (implicit ec: ExecutionContext) {
 
+  private def getPaymentsOnAccountFromDocumentDetails(documentDetails: List[DocumentDetail])
+                                                     (implicit user: MtdItUser[_], hc: HeaderCarrier): Either[Throwable, PaymentOnAccount] = {
+    {
+      for {
+        poaOneDocDetail   <- documentDetails.filter(_.isPoAOne).sortBy(_.taxYear).reverse.headOption
+        transactionId      = poaOneDocDetail.transactionId
+        poaOneTaxYear      = makeTaxYearWithEndYear(poaOneDocDetail.taxYear)
+        poaOneTotalAmount  = poaOneDocDetail.originalAmount
+        poaTwoDocDetail   <- documentDetails.filter(_.isPoATwo).sortBy(_.taxYear).reverse.headOption
+        poaTwoTaxYear      = makeTaxYearWithEndYear(poaTwoDocDetail.taxYear)
+        poaTwoTotalAmount  = poaTwoDocDetail.originalAmount
+        if poaOneTaxYear  == poaTwoTaxYear
+      } yield {
+        // TODO: .get has to be used here, DocumentDetail.originalAmount is not an optional field according to #API1553 - this needs to be fixed in BE/FE
+        PaymentOnAccount(
+          transactionId,
+          poaOneTaxYear,
+          poaOneTotalAmount.get,
```

Comment:
might be worth to set relevant fields in the model as optional; but yes, we should avoid calling .get